lib/jsonwp-proxy/proxy.js

```diff
@@ -131,9 +131,15 @@ class JWProxy {
       timeout: this.timeout,
       forever: this.keepAlive,
     };
-    if (body !== null) {
+    if (util.hasValue(body)) {
       if (typeof body !== 'object') {
-        body = JSON.parse(body);
+        try {
+          body = JSON.parse(body);
+        } catch (e) {
+          throw new Error('Cannot interpret the request body as valid JSON: ' +
+            _.truncate(_.isString(body) ? body : JSON.stringify(body),
+            {length: LOG_OBJ_LENGTH}));
```

Comment:
Could you please not truncate this if it was in debug or verbose mode.

Reply:
which driver property points to the fact that debug mode is enabled? I don't know such

Reply:
https://github.com/appium/appium-base-driver/pull/72/files#issuecomment-245351191

Reply:
That PR keep truncate and I know why. Removing it completely will quickly overload the log buffer with megabytes-long strings and cause an out of memory error